nixos/modules/services/web-servers/ttyd.nix

```diff
@@ -0,0 +1,202 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.ttyd;
+
+  # Command line arguments for the ttyd daemon
+  args = [ "--port ${toString cfg.port}" ]
+         ++ optional (cfg.socket != null) "--interface ${cfg.socket}"
```

Comment:
Then you should also use `optionals (cfg.socket != null) [ "--interface" cfg.socket ]` here so the escaping is done per-argument. Same for all other arguments.

Reply:
Yup, much better. Thanks. Fixed.